### PR TITLE
Lookout UI Job Cancellation

### PR DESCRIPTION
**Pull request by carlocamurri**
Allows users to cancel jobs from the Lookout UI.

Note that this requires CORS to be enabled on the Armada API.

image

image

### CODE REVIEW

**Review by jankaspar — CHANGES_REQUESTED** [review]
Looks alright except the CORS policy.

File: internal/common/grpc/gateway.go [hunk 1]
```diff
@@ -75,6 +75,26 @@ func CreateGatewayHandler(
 	}
 }
 
+func allowCORS(h http.Handler) http.Handler {
```
[the remark this answers is not in the document]
Comment (devlounge) [reply internal/common/grpc/gateway.go]: I think that this cancellation popup contains too much the word "cancel"

Cancel Jobs
Cancel the following jobs
CANCEL JOBS

**Comment by devlounge** [comment]
I think that this cancellation popup contains too much the word "cancel"

Cancel Jobs
Cancel the following jobs
CANCEL JOBS

File: config/armada/config.yaml [hunk 2]
```diff
@@ -2,6 +2,7 @@
 grpcPort: 50051
 httpPort: 8080
 metricsPort: 9000
+corsAllowedOrigins: ["http://localhost:3000"]
```
Comment (jankaspar) [inline config/armada/config.yaml:5]: I would prefer default to be empty and set inside helm chart to lookout domain. It is minor but we don't want any localhost to talk to the api.

**Review by jankaspar — APPROVED** [review]

**Pull request merged**